src/components/Loading.js

```diff
@@ -0,0 +1,42 @@
+import { dispatch } from "utils/dispatch";
+import {
+  // defaultMapConfig,
+  TOTAL_NUMBER_OF_MAP_LAYERS,
+} from "utils/constants";
+
+export class Loading {
+  //laodng screen container
+  loadingScreen = document.getElementById("loading-container");
+
+  // loaded layers counter
+  layersLoaded = 0;
+
+  // optionally pass in the mapconfig to keep
```

Comment:
Currently the way in which new layers are added to the map happens in such a way that if a layer is set to `false` in the `mapConfig` a `fetch` request will still be made for that layer. This means that the map will still be waiting for all layers to finish fetching, because even if they're not added to the map they are still cached in the map's `dataLayers`. It might be better to only `fetch` a layer's data if it is enabled when the map loads, and then when a layer is toggled on for the first time, fetch the data. 

Anyway, I bring this up as I'm not sure if that's what you're already thinking about / planning for. It's my preference to not leave dead code in for future work as to not confuse others.

Reply:
That about what I was thinking.  Yeah I noticed.  With changes mentioned from above this will be addressed.